sqlmesh/core/plan/definition.py

```diff
@@ -87,7 +87,7 @@ def __init__(
         self.categorizer_config = categorizer_config or CategorizerConfig()
         self.auto_categorization_enabled = auto_categorization_enabled
         self._start = start if start or not (is_dev and forward_only) else yesterday_ds()
-        self._end = end if end or not is_dev else now()
+        self._end = end
```

Comment:
Please note that no longer setting the end in the constructor impacts the end we set in the environment: https://github.com/TobikoData/sqlmesh/blob/48cc0be158c379f49cd72704cc1c3600cc23b27d/sqlmesh/core/plan/definition.py#L229. Please also note that we may expect end to be None in the environment in some cases.